work12/index.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+const wrap = document.getElementById("js-wrap")
+const ul = document.getElementById("js-list");
+const button = document.getElementById("js-button");
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/bu5z";
+const url = "https://myjson.dit.upm.es/api/bins/2hj3";
+
+async function getData() {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      const json = await response.json();
+      return json;
+    } else {
+      throw new Error(response.statusText);
+    }
+  } catch (e) {
+    throw new Error(e);
+  }
+}
```

Comment:
I fixed it to console.error (e).


I searched for "console.error (e)" and "throw".

For the getData function
If an error occurs with "try", the value will be passed to "catch" and the process will stop here.
So you don't have to use "throw"
That's right?

commit:  c230124

Reply:
Yeah, every try...catch statement is like that. I've also found other reference, so I would share it to you.
[制御フローとエラー処理](https://developer.mozilla.org/ja/docs/Web/JavaScript/Guide/Control_flow_and_error_handling)
Under the try...catch statement part explains, please read it if you need.
This page also taught me `When logging errors to the console inside a catch block, using "console.error()" rather than "console.log()" is advised for debugging.` as the best practice!

Reply:
@yuka830 
I checked the reference.
"If you have other catch statements outside, you can use throw in the catch statement."
Is it explained here? .. ([reference](https://developer.mozilla.org/ja/docs/Web/JavaScript/Guide/Control_flow_and_error_handling#:~:text=finally%20%E3%83%96%E3%83%AD%E3%83%83%E3 ％82％AF％E3％81％AB％E3％82％88％E3％82％8B％E8％BF％94％E5％80％A4％E3％81％AE％E4％B8％8A％E6％9B ％B8％E3％81％8D％E3％81％AF％E3％80％81catch％20％E3％83％96％E3％83％AD％E3％83％83％E3％82％AF％E5％86 ％85％E3％81％A7％E7％99％BA％E7％94％9F％E3％81％97％E3％81％9F％E3％80％81％E3％81％BE％E3％81％9F ％E3％81％AF％E5％86％8D％E7％99％BA％E7％94％9F％E3％81％97％E3％81％9F％E4％BE％8B％E5％A4％96％E3 ％81％AB％E3％82％82％E9％81％A9％E7％94％A8％E3％81％95％E3％82％8C％E3％81％BE％E3％81％99％E3％80 ％82))

Reply:
Yes, that's right.
But I'm sorry, I taught you wrong thing here. 
Actually, you can use `throw` in this part, because this `getData` is called inside `getListData`, and `getListData` has `catch` statement.... I missed about this point.
I'm sorry about that 😭 😭

Reply:
Thanks to your advice, I learned a lot.😊

So it's strange to write a throw statement in a catch block without using an external catch statement, right?

Also, this time, if an error is thrown by getData (), the catch block of getListData () will work, right?

I'm leaving a comment in the try / catch block for my understanding. please forgive me

commit:[e538737](https://github.com/yasu-pro/js-handson/pull/12/commits/e538737a1650a589532be6bead00dddb8a12ee68)